test/junit/scala/PartialFunctionCompositionTests.scala

```diff
@@ -0,0 +1,139 @@
+package scala
+
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(classOf[JUnit4])
+class PartialFunctionCompositionTests {
+  val fallbackFun = (_: String) => "fallback"
+  val passAll: PartialFunction[String, String] = { case any => any }
+  val passShort: PartialFunction[String, String] = { case any if any.length < 5 => any }
+  val passPass: PartialFunction[String, String] = { case s if s.contains("pass") => s }
+
```

Comment:
Should probably add side effect cases like:

```var foo = 0
val passAllWithSideEffect: PartialFunction[String, String] = { case any => foo+=1; any}
```

and in the assert section `assert(foo == xxx)`